lib/hawkbit/hawkbit.c

```diff
@@ -0,0 +1,1212 @@
+/*
+ * Copyright (c) 2016-2017 Linaro Limited
+ * Copyright (c) 2018 Open Source Foundries Limited
+ * Copyright (c) 2018 Foundries.io
+ * Copyright (c) 2020 Linumiz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(hawkbit);
+
+#include <stdio.h>
+#include <zephyr.h>
+#include <string.h>
+#include <stdlib.h>
+#include <fs/nvs.h>
+#include <data/json.h>
+#include <net/net_ip.h>
+#include <net/socket.h>
+#include <net/net_mgmt.h>
+#include <power/reboot.h>
+#include <drivers/flash.h>
+#include <net/http_client.h>
+#include <net/dns_resolve.h>
+#include <logging/log_ctrl.h>
+
+#include "hawkbit_priv.h"
+#include "hawkbit_device.h"
+#include "include/hawkbit.h"
+#include "hawkbit_firmware.h"
+
+#if defined(CONFIG_NET_SOCKETS_SOCKOPT_TLS)
+#define CA_CERTIFICATE_TAG 1
+#include <net/tls_credentials.h>
+#endif
+
+#define ADDRESS_ID 1
+
+#define CANCEL_BASE_SIZE 50
+#define RECV_BUFFER_SIZE 640
+#define URL_BUFFER_SIZE 300
+#define STATUS_BUFFER_SIZE 200
+#define DOWNLOAD_HTTP_SIZE 200
+#define DEPLOYMENT_BASE_SIZE 50
+#define NETWORK_TIMEOUT K_SECONDS(2)
+#define HAWKBIT_RECV_TIMEOUT K_SECONDS(300)
+
+#define FLASH_BANK_SIZE DT_FLASH_AREA_IMAGE_1_SIZE
+#define HTTP_HEADER_CONTENT_TYPE_JSON "application/json;charset=UTF-8"
+
+#if ((CONFIG_HAWKBIT_POLL_INTERVAL > 1)\
+		&& (CONFIG_HAWKBIT_POLL_INTERVAL < 43200))
+static u32_t poll_sleep = K_SECONDS(CONFIG_HAWKBIT_POLL_INTERVAL * 60);
+#else
+static u32_t poll_sleep = K_SECONDS(300);
+#endif
+
+static struct nvs_fs fs;
+u8_t response_data[1100];
```

Comment:
This should be static. Could you also use a define for the value, we try to avoid magic constants.
Could these two variables put to `hawkbit_context`?

Reply:
can't add nvs_fs to hawkbit_context. during every poll operation hawkbit_context need to memset to zero.